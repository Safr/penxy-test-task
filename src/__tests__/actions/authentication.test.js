import {
  notAuthed, loginSuccess, registerSuccess, update,
} from '../../actions';

describe('Authentication Actions', () => {
  const data = { name: 'name', email: 'email', access_token: 'token' };
  test('should login and get user data', () => {
    const action = loginSuccess(data);
    expect(action).toMatchSnapshot();
  });
  test('should unauthenticate', () => {
    const action = notAuthed();
    expect(action).toMatchSnapshot();
  });
  test('should register and get user data', () => {
    const action = registerSuccess(data);
    expect(action).toMatchSnapshot();
  });
  test('should update user data', () => {
    const action = update({ name: 'name', email: 'email' });
    expect(action).toMatchSnapshot();
  });
});
