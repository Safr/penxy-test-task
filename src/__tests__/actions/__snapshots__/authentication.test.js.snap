// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication Actions should login and get user data 1`] = `
Object {
  "access_token": "token",
  "email": "email",
  "name": "name",
  "type": "LOGIN_SUCCESS",
}
`;

exports[`Authentication Actions should register and get user data 1`] = `
Object {
  "access_token": "token",
  "email": "email",
  "name": "name",
  "type": "REGISTER_SUCCESS",
}
`;

exports[`Authentication Actions should unauthenticate 1`] = `
Object {
  "type": "NOT_AUTHED",
}
`;

exports[`Authentication Actions should update user data 1`] = `
Object {
  "email": "email",
  "name": "name",
  "type": "UPDATE",
}
`;
